tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -70,6 +67,7 @@ syntaxDatatype language datatype = case datatype of
     name = toName (datatypeNameStatus datatype) (getDatatypeName (TreeSitter.Deserialize.datatypeName datatype))
     deriveClause = [ DerivClause Nothing [ ConT ''TS.Unmarshal, ConT ''Eq, ConT ''Ord, ConT ''Show, ConT ''Generic ] ]
     generatedDatatype name cons typeParameterName = DataD [] name [PlainTV typeParameterName] Nothing cons deriveClause
+    typeParameterName = mkName "a"
```

Comment:
Careful: remember that `mkName` and `newName` have different semantics.

Reply:
Corrected! See https://github.com/tree-sitter/haskell-tree-sitter/pull/181#discussion_r322298283 above.